app/models/study.rb

```diff
@@ -209,15 +211,26 @@ def non_primary_data
 
   # External Resource links
   has_many :external_resources, as: :resource_links, dependent: :destroy
+  accepts_nested_attributes_for :external_resources, allow_destroy: true
 
-  # Study Detail (full html description)
-  has_one :study_detail, dependent: :delete_all
+  # Anonymous Reviewer Access
+  has_one :reviewer_access, dependent: :delete_all
+  accepts_nested_attributes_for :reviewer_access, allow_destroy: true
 
   # DownloadAgreement (extra user terms for downloading data)
   has_one :download_agreement, dependent: :delete_all
+  accepts_nested_attributes_for :download_agreement, allow_destroy: true
 
-  # Anonymous Reviewer Access
-  has_one :reviewer_access, dependent: :delete_all
+  # Study Detail (full html description)
+  #
+  # NOTE: THIS ASSOCIATION MUST ALWAYS BE DECLARED LAST
+  #
+  # This is due to the after_save :set_study_description_text callback which will cause the parent study
+  # to update itself with the plain-text version of the description
+  # this will clobber any incoming changes to other associations, since the study is always saved first
+  # therefore, put any new associations above :study_detail
```

Comment:
Good comment.  I just spent a few minutes trying to think if there was a cleaner way to structure the updates so we wouldn't have this order-specific dependency.  

Would it work if we updated `set_study_description_text` in StudyDetail to something like:
```
  study_object = Study.find(self.study_id)
  text_description = self.plain_text_description
  study_object.update!(description: text_description)
  self.study.description = text_description # make sure the update above isn't clobbered by a study save if this is happening in a hook
```
By avoiding the reload and save, and just updating the object in the DB directly instead of off the association, we might be able to avoid gotchas like this in the future.  But I'm not sure this will work

Reply:
Well whaddya know... that works like a charm.  Nice catch.